Build error with nginx 1.23.0

Build with nginx 1.23.0 occurs error like below.

```
#7 37.04 make -f objs/Makefile modules
#7 37.04 make[1]: Entering directory '/usr/local/src/nginx-1.23.0'
#7 37.04 cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/core -I src/event -I src/event/modules -I src/os/unix -I -pipe -I -O -I -W -I -Wall -I -Wpointer-arith -I -Wno-unused-parameter -I -Werror -I -g -I /usr/local/src/ngx_mruby/mruby/src -I /usr/local/src/ngx_mruby/mruby/include -I objs -I src/http -I src/http/modules \
#7 37.04        -o objs/addon/http/ngx_http_mruby_module.o \
#7 37.04        ../ngx_mruby/src/http/ngx_http_mruby_module.c
#7 37.25 cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/core -I src/event -I src/event/modules -I src/os/unix -I -pipe -I -O -I -W -I -Wall -I -Wpointer-arith -I -Wno-unused-parameter -I -Werror -I -g -I /usr/local/src/ngx_mruby/mruby/src -I /usr/local/src/ngx_mruby/mruby/include -I objs -I src/http -I src/http/modules \
#7 37.25        -o objs/addon/http/ngx_http_mruby_core.o \
#7 37.25        ../ngx_mruby/src/http/ngx_http_mruby_core.c
#7 37.36 cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/core -I src/event -I src/event/modules -I src/os/unix -I -pipe -I -O -I -W -I -Wall -I -Wpointer-arith -I -Wno-unused-parameter -I -Werror -I -g -I /usr/local/src/ngx_mruby/mruby/src -I /usr/local/src/ngx_mruby/mruby/include -I objs -I src/http -I src/http/modules \
#7 37.36        -o objs/addon/http/ngx_http_mruby_init.o \
#7 37.36        ../ngx_mruby/src/http/ngx_http_mruby_init.c
#7 37.41 cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/core -I src/event -I src/event/modules -I src/os/unix -I -pipe -I -O -I -W -I -Wall -I -Wpointer-arith -I -Wno-unused-parameter -I -Werror -I -g -I /usr/local/src/ngx_mruby/mruby/src -I /usr/local/src/ngx_mruby/mruby/include -I objs -I src/http -I src/http/modules \
#7 37.41        -o objs/addon/http/ngx_http_mruby_request.o \
#7 37.41        ../ngx_mruby/src/http/ngx_http_mruby_request.c
#7 37.55 cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/core -I src/event -I src/event/modules -I src/os/unix -I -pipe -I -O -I -W -I -Wall -I -Wpointer-arith -I -Wno-unused-parameter -I -Werror -I -g -I /usr/local/src/ngx_mruby/mruby/src -I /usr/local/src/ngx_mruby/mruby/include -I objs -I src/http -I src/http/modules \
#7 37.55        -o objs/addon/http/ngx_http_mruby_var.o \
#7 37.55        ../ngx_mruby/src/http/ngx_http_mruby_var.c
#7 37.64 cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/core -I src/event -I src/event/modules -I src/os/unix -I -pipe -I -O -I -W -I -Wall -I -Wpointer-arith -I -Wno-unused-parameter -I -Werror -I -g -I /usr/local/src/ngx_mruby/mruby/src -I /usr/local/src/ngx_mruby/mruby/include -I objs -I src/http -I src/http/modules \
#7 37.64        -o objs/addon/http/ngx_http_mruby_connection.o \
#7 37.64        ../ngx_mruby/src/http/ngx_http_mruby_connection.c
#7 37.68 In file included from src/event/ngx_event.h:526,
#7 37.68                  from src/http/ngx_http_upstream.h:14,
#7 37.68                  from src/http/ngx_http.h:34,
#7 37.68                  from ../ngx_mruby/src/http/ngx_http_mruby_request.h:11,
#7 37.68                  from ../ngx_mruby/src/http/ngx_http_mruby_connection.c:9:
#7 37.68 src/event/ngx_event_udp.h:38:27: error: field 'pkt6' has incomplete type
#7 37.68    38 |     struct in6_pktinfo    pkt6;
#7 37.68       |                           ^~~~
#7 37.68 make[1]: Leaving directory '/usr/local/src/nginx-1.23.0'
#7 37.68 make[1]: *** [objs/Makefile:1304: objs/addon/http/ngx_http_mruby_connection.o] Error 1
#7 37.68 make: *** [Makefile:16: modules] Error 2
```

## Step to reproduce

- Edit `NGINX_SRC_MINOR` in `nginx_version` file to `23`
- Execute `docker build -t local/ngx_mruby . --no-cache` on root of this repository.

--- a/src/http/ngx_http_mruby_async.h
+++ b/src/http/ngx_http_mruby_async.h
@@ -8,6 +8,7 @@
 #define NGX_HTTP_MRUBY_ASYNC_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_http.h>
 
 #include <mruby.h>

--- a/src/http/ngx_http_mruby_connection.h
+++ b/src/http/ngx_http_mruby_connection.h
@@ -7,6 +7,10 @@
 #ifndef NGX_HTTP_MRUBY_CONNECTION_H
 #define NGX_HTTP_MRUBY_CONNECTION_H
 
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+
 #include <mruby.h>
 
 void ngx_mrb_conn_class_init(mrb_state *mrb, struct RClass *class);

--- a/src/http/ngx_http_mruby_core.h
+++ b/src/http/ngx_http_mruby_core.h
@@ -8,6 +8,7 @@
 #define NGX_HTTP_MRUBY_CORE_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_http.h>
 
 #include <mruby.h>

--- a/src/http/ngx_http_mruby_filter.h
+++ b/src/http/ngx_http_mruby_filter.h
@@ -7,6 +7,10 @@
 #ifndef NGX_HTTP_MRUBY_FILTER_H
 #define NGX_HTTP_MRUBY_FILTER_H
 
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+
 #include <mruby.h>
 
 void ngx_mrb_filter_class_init(mrb_state *mrb, struct RClass *class);

--- a/src/http/ngx_http_mruby_init.h
+++ b/src/http/ngx_http_mruby_init.h
@@ -8,6 +8,7 @@
 #define NGX_HTTP_MRUBY_INIT_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_http.h>
 
 #include <mruby.h>

--- a/src/http/ngx_http_mruby_request.h
+++ b/src/http/ngx_http_mruby_request.h
@@ -8,6 +8,7 @@
 #define NGX_HTTP_MRUBY_REQUEST_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_http.h>
 
 #include <mruby.h>

--- a/src/http/ngx_http_mruby_server.h
+++ b/src/http/ngx_http_mruby_server.h
@@ -7,6 +7,10 @@
 #ifndef NGX_HTTP_MRUBY_SERVER_H
 #define NGX_HTTP_MRUBY_SERVER_H
 
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+
 #include <mruby.h>
 
 void ngx_mrb_server_class_init(mrb_state *mrb, struct RClass *class);

--- a/src/http/ngx_http_mruby_ssl.h
+++ b/src/http/ngx_http_mruby_ssl.h
@@ -8,6 +8,8 @@
 #define NGX_HTTP_MRUBY_SSL_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
 
 #if (NGX_HTTP_SSL)
 

--- a/src/http/ngx_http_mruby_upstream.h
+++ b/src/http/ngx_http_mruby_upstream.h
@@ -7,7 +7,10 @@
 #ifndef NGX_HTTP_MRUBY_UPSTREAM_H
 #define NGX_HTTP_MRUBY_UPSTREAM_H
 
-#include <nginx.h>
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+
 #if (nginx_version > 1007999)
 #define NGX_USE_MRUBY_UPSTREAM
 #endif
@@ -19,8 +22,6 @@
 void ngx_mrb_upstream_class_init(mrb_state *mrb, struct RClass *class);
 
 #ifdef NGX_HTTP_MRUBY_UPSTREAM_C
-
-#include <ngx_config.h>
 
 /* reference from src/http/modules/ngx_http_upstream_keepalive_module.c */
 

--- a/src/http/ngx_http_mruby_var.h
+++ b/src/http/ngx_http_mruby_var.h
@@ -8,6 +8,7 @@
 #define NGX_HTTP_MRUBY_VAR_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_http.h>
 
 #include <mruby.h>

--- a/src/stream/ngx_stream_mruby_async.h
+++ b/src/stream/ngx_stream_mruby_async.h
@@ -8,6 +8,7 @@
 #define NGX_STREAM_MRUBY_ASYNC_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_stream.h>
 
 #include <mruby.h>

--- a/src/stream/ngx_stream_mruby_connection.h
+++ b/src/stream/ngx_stream_mruby_connection.h
@@ -7,6 +7,10 @@
 #ifndef NGX_STREAM_MRUBY_CONNECTION_H
 #define NGX_STREAM_MRUBY_CONNECTION_H
 
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_stream.h>
+
 #include <mruby.h>
 
 void ngx_stream_mrb_conn_class_init(mrb_state *mrb, struct RClass *class);

--- a/src/stream/ngx_stream_mruby_core.h
+++ b/src/stream/ngx_stream_mruby_core.h
@@ -8,6 +8,7 @@
 #define NGX_STREAM_MRUBY_CORE_H
 
 #include <ngx_config.h>
+#include <ngx_core.h>
 #include <ngx_stream.h>
 
 #include <mruby.h>

--- a/src/stream/ngx_stream_mruby_init.h
+++ b/src/stream/ngx_stream_mruby_init.h
@@ -9,6 +9,7 @@
 
 #include <ngx_config.h>
 #include <ngx_core.h>
+#include <ngx_stream.h>
 
 #include <mruby.h>
 

--- a/src/stream/ngx_stream_mruby_module.h
+++ b/src/stream/ngx_stream_mruby_module.h
@@ -7,7 +7,7 @@
 #ifndef NGX_STREAM_MRUBY_MODULE_H
 #define NGX_STREAM_MRUBY_MODULE_H
 
-#include <nginx.h>
+#include <ngx_config.h>
 #include <ngx_core.h>
 #include <ngx_stream.h>
 

